src/test/kotlin/sirius/db/mongo/MongoFilterFactoryTest.kt

```diff
@@ -8,355 +8,543 @@
 
 package sirius.db.mongo
 
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
 import sirius.db.mixing.Mapping
 import sirius.db.mongo.properties.MongoStringListEntity
-import sirius.kernel.BaseSpecification
+import sirius.kernel.SiriusExtension
 import sirius.kernel.commons.Value
 import sirius.kernel.di.std.Part
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
 
-class MongoFilterFactorySpec extends BaseSpecification {
+@ExtendWith(SiriusExtension::class)
+class MongoFilterFactoryTest {
+    companion object {
+        @Part
+        private lateinit var mango: Mango
 
-    @Part
-    private static Mango mango
+        @Part
+        private lateinit var mongo: Mongo
 
-            @Part
-            private static Mongo mongo
+        private fun prefixSearch(query: String): Optional<Doc>? {
+            return mongo.find().where(QueryBuilder.FILTERS.prefix(PrefixTestEntity.PREFIX, query))
+                    .singleIn(PrefixTestEntity::class.java)
+        }
 
-            private Optional<PrefixTestEntity> prefixSearch(String query) {
-        return mongo.find().where(QueryBuilder.FILTERS.prefix(PrefixTestEntity.PREFIX, query))
-                .singleIn(PrefixTestEntity.class)
+        private fun textSearch(query: String): Optional<Doc>? {
+            return mongo.find().where(QueryBuilder.FILTERS.text(query))
+                    .singleIn(PrefixTestEntity::class.java)
+        }
     }
 
-    private Optional<PrefixTestEntity> textSearch(String query) {
-        return mongo.find().where(QueryBuilder.FILTERS.text(query))
-                .singleIn(PrefixTestEntity.class)
-    }
+    @Test
+    fun `prefix search works`() {
+        val prefixTestEntity = PrefixTestEntity()
+        prefixTestEntity.prefix = "test-1"
+        mango.update(prefixTestEntity)
 
-    def "prefix search works"() {
-        when:
-        PrefixTestEntity test = new PrefixTestEntity()
-        test.setPrefix("test-1")
-        mango.update(test)
-        then:
-        prefixSearch("te").isPresent()
-        and:
-        prefixSearch("test-").isPresent()
-        and:
-        prefixSearch("Test-1").isPresent()
-        and:
-        textSearch("Test-1").isPresent()
-        and:
-        textSearch("Test").isPresent()
-        and:
-        textSearch("test-1").isPresent()
-        and:
-        !textSearch("te").isPresent()
+        assertTrue { prefixSearch("te")!!.isPresent }
```

Comment:
With `Doc?` this would turn into a `!= null` or `assertNotNull` if thats available